### PR TITLE
Opt in deprecations docs

### PR DESCRIPTION
**Pull request by greg0ire**

### CODE REVIEW

File: docs/en/reference/upgrading.rst [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    );
+
+Since this error is silenced, it will not produce any effect unless you
+opt-in by setting up an error handler designed to ignore the silence
+operator in that case. Such an error handler could look like this:
+
+.. code-block:: php
+
+    <?php
+    set_error_handler(function (
```
Comment (Majkl578) [inline docs/en/reference/upgrading.rst]: Is it actually a good idea to overwrite handler for all messages like that? Can we maybe suggest this instead only for deprecations?
```php
set_error_handler(
    function (int $code, string $message, string $file, int $line) : void {
        echo sprintf('Deprecation: "%s" in %s:%d', $message, $file, $line);
    },
    E_USER_DEPRECATED
);
```
https://3v4l.org/AFfPk

File: docs/en/reference/upgrading.rst [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            return;
+        }
+
+        // normal error handling here
+    });
+
+This is of course overly simplified, and if you are looking for such an
+error handler, consider the ``symfony/debug``, error handler that will
+log deprecations. You may also be interested by the
+``symfony/phpunit-bridge`` error handler that will display deprecations
```
Comment (Majkl578) [inline docs/en/reference/upgrading.rst]: It'd be nice to expand on the `symfony/phpunit-bridge` part. Imho that's the most convenient opt-in method.

File: docs/en/reference/upgrading.rst (hunk 1 above)
Comment (greg0ire) [reply docs/en/reference/upgrading.rst]: Fixed

**Review by jwage — APPROVED** [review]

**Comment by Majkl578** [comment]
Uh, we don't usually merge to 2.x branches, rather cherry-pick from master. This needs to be picked up to master now. :|

**Comment by greg0ire** [comment]
> Uh, we don't usually merge to 2.x branches, rather cherry-pick from master. This needs to be picked up to master now. :|

Ah right, sorry for this

**Pull request merged**